R/calculate_within_batch_ari.R

```diff
@@ -48,54 +48,96 @@ calculate_within_batch_ari <- function(individual_sce_list,
   }
 
   # Check that `pc_name` is in merged SCE object
-  if (!pc_name %in% reducedDimNames(merged_sce)) {
-    stop("The provided `pc_name` cannot be found in the `merged_sce`.")
+  if (!any(pc_names %in% reducedDimNames(merged_sce))) {
```

Comment:
If we want to check that they are all present, we need "not all"
```suggestion
  if (!all(pc_names %in% reducedDimNames(merged_sce))) {
```